staking/programs/integrity-pool/src/lib.rs

```diff
@@ -250,4 +251,30 @@ pub mod integrity_pool {
         delegation_record.advance(get_current_epoch()?)?;
         Ok(())
     }
+
+    pub fn create_slash_event(
+        ctx: Context<CreateSlashEvent>,
+        index: u64,
+        slash_ratio: frac64,
+        slash_custody: Pubkey,
```

Comment:
I think you should probably check that this is a Pyth token account, you can do that by putting it in the context instead of putting it in the instruction arguments

Reply:
fixed.